.github/workflows/ci.yml

```diff
@@ -22,8 +26,39 @@ jobs:
     - name: Install Newman dependency
       run: npm install --prefix tests/postman/
 
-    - name: Run the API and Postman's tests
-      run: |
-        cd tests/postman/ && newman run *
+    - name: Start Netlify Dev for local function testing
+      run:  npm run netlify-dev & sleep 10 && cd tests && node newman.js 
+      env:
+        STRIPE_TEST_KEY: sk_test_7LEesZrYrgDtyxsc6Hy5i0lS00GVFEbmqb
+        ENDPOINT: http://localhost:5050
+
+    - name: Run tests against deployed preview
+      run:  cd tests && node newman.js 
       env:
-        CI: true
+        STRIPE_TEST_KEY: sk_test_7LEesZrYrgDtyxsc6Hy5i0lS00GVFEbmqb
+        ENDPOINT: https://deploy-preview-${{github.event.number}}--ambianic-serverless.netlify.app/.netlify/functions
```

Comment:
@vickywane how do you know the netlify deployment preview is ready? [See how we use the wait step](https://github.com/ambianic/ambianic-ui/blob/0c523c39d4eeca2b820b0d0323236cd923d497ac/.github/workflows/lighthouse.yml#L19) in the UI deployment preview tests.

Reply:
There's a bug with `jakepartusch/wait-for-netlify-action@v1`, I had it set it up previously.

There's a mismatch in the PR number which leads it to continuously ping an invalid URL before timing out.

Reply:
The ping functionality has been written in `scripts.sh` file. It would continuously ping the deployed preview URL for a 200 response code with a set timeout.